src/Microsoft.AspNetCore.ResponseCaching/ResponseCache/MemoryResponseCache.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Microsoft.AspNetCore.ResponseCaching
+{
+    public class MemoryResponseCache : IResponseCache
+    {
+        private readonly IMemoryCache _cache;
+
+        public MemoryResponseCache(IMemoryCache cache)
+        {
+            if (cache == null)
+            {
+                throw new ArgumentNullException(nameof(cache));
+            }
+
+            _cache = cache;
+        }
+
+        public object Get(string key)
+        {
+            try
+            {
+                return _cache.Get(key);
+            }
+            catch
```

Comment:
These methods also throw exceptions, for example, if the underlying storage has been disposed.